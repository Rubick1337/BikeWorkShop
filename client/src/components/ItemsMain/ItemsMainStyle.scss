.container__item
{
  display: flex;
  padding: 20px 30px;
  flex-direction: column;
  align-items: center;
  gap: 30px;
}
.img__bike
{
  background-image: url(../../images/image3.png);
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  width: 435px;
  height: 270px;
}
.img__part{
  background-image: url(../../images/free-icon-repair-tool-2760290.png);
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  width: 315.352px;
  height: 270px;
}
.img__const{
  background-image: url(../../images/free-icon-blueprint-671593.png);
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  width: 270px;
  height: 270px;
}
.button__sell{
  background-image: url(../../images/Group1.jpg);
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  width: 33px;
  height: 33px;
  transition: transform 0.3s ease;
}
.button__sell__left
{
  background-image: url(../../images/Group2.png);
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  width: 33px;
  height: 33px;
  transition: transform 0.3s ease;
}
.button__sell:hover {
  transform: translateX(20px);
}


.button__sell__left:hover {
  transform: translateX(-20px);
}


.button__sell, .button__sell__left {
  transition: transform 0.3s ease;
}
.container__sell__item
{
  padding: 30px 0px;
  gap: 20.092px;
  display: flex;
  justify-content: center;
  align-items: center;
}
.container__img__button
{
  display: flex;
  align-items: center;
  gap: 15px;
}
.text__sell h3
{
  color: #000;
  font-family: Forum;
  font-size: 24px;
  font-style: normal;
  font-weight: 400;
  line-height: 40px; /* 166.667% */
}
@media (max-width: 840px) {
  .img__bike
  {
    width: 300px;
    height: 186.207px;
  }
  .img__part
  {
    width: 217.487px;
    height: 186.21px;
  }
  .img__const
  {
    width: 186.21px;
    height: 186.21px;
  }
}
@media (max-width: 700px) {
  .container__sell__item
  {
    flex-wrap: wrap;
  }
  .reverse
  {
    flex-direction: column-reverse;
  }
  .img__bike
  {
    width: 217px;
    height: 134.69px;
  }
}