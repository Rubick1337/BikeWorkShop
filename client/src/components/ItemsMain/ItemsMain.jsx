import React from "react";
import "./ItemsMainStyle.scss"
export  default  function ItemsMain()
{
    return (
        <div className="container__item">
            <div className="title_section">
                <h3>Наш ассортимент</h3>
                <div className="line__title"></div>
            </div>
            <div className="container__sell">
                <div className="container__sell__item">
                    <div className="container__img__button">
                        <div className="img__bike"></div>
                        <div className="button__sell"></div>
                    </div>
                    <div className="text__sell">
                        <h3>Откройте для себя мир приключений с нашими велосипедами! Ищете идеальный велосипед для
                            города или экстремальных поездок по горам? У нас есть всё, что вам нужно. Наш ассортимент
                            сочетает в себе передовые технологии и стильный дизайн.
                            Переходите на страницу ассортимента, чтобы найти свой идеальный велосипед и начать новое
                            путешествие!</h3>
                    </div>
                </div>
                <div className="container__sell__item reverse">
                    <div className="text__sell">
                        <h3>Улучшите свой велосипед с нашими высококачественными деталями!
                            Ищете надежные компоненты для улучшения производительности и комфорта вашего велосипеда?
                            Наш ассортимент деталей включает в себя всё: от прочных рам до высокоточных тормозов.
                            Посетите нашу страницу ассортимента и найдите идеальные решения для любых условий и стиля езды!</h3>
                    </div>
                    <div className="container__img__button">
                        <div className="button__sell__left"></div>
                        <div className="img__part"></div>
                    </div>
                </div>
                <div className="container__sell__item">
                    <div className="container__img__button">
                        <div className="img__const"></div>
                        <div className="button__sell"></div>
                    </div>
                    <div className="text__sell">
                        <h3>Создайте велосипед своей мечты с нашим конструктором!
                            Хотите уникальный велосипед, который отражает ваш стиль и нужды?
                            Наш конструктор позволяет выбрать все — от рамы до последних штрихов.
                            Погрузитесь в процесс создания и получите велосипед, который будет идеально соответствовать вашим ожиданиям.
                            Перейдите на страницу конструктора и начните своё уникальное путешествие!</h3>
                    </div>
                </div>
            </div>
        </div>
    )
}