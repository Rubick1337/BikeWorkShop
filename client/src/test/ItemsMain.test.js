import React from 'react';
import { render } from '@testing-library/react';
import ItemsMain from '../components/ItemsMain/ItemsMain'; // Убедитесь, что путь правильный

describe('ItemsMain Component', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(<ItemsMain />);
        expect(asFragment()).toMatchSnapshot();
    });
});