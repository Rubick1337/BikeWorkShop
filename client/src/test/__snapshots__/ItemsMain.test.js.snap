// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsMain Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container__item"
  >
    <div
      class="title_section"
    >
      <h3>
        Наш ассортимент
      </h3>
      <div
        class="line__title"
      />
    </div>
    <div
      class="container__sell"
    >
      <div
        class="container__sell__item"
      >
        <div
          class="container__img__button"
        >
          <div
            class="img__bike"
          />
          <div
            class="button__sell"
          />
        </div>
        <div
          class="text__sell"
        >
          <h3>
            Откройте для себя мир приключений с нашими велосипедами! Ищете идеальный велосипед для города или экстремальных поездок по горам? У нас есть всё, что вам нужно. Наш ассортимент сочетает в себе передовые технологии и стильный дизайн. Переходите на страницу ассортимента, чтобы найти свой идеальный велосипед и начать новое путешествие!
          </h3>
        </div>
      </div>
      <div
        class="container__sell__item reverse"
      >
        <div
          class="text__sell"
        >
          <h3>
            Улучшите свой велосипед с нашими высококачественными деталями! Ищете надежные компоненты для улучшения производительности и комфорта вашего велосипеда? Наш ассортимент деталей включает в себя всё: от прочных рам до высокоточных тормозов. Посетите нашу страницу ассортимента и найдите идеальные решения для любых условий и стиля езды!
          </h3>
        </div>
        <div
          class="container__img__button"
        >
          <div
            class="button__sell__left"
          />
          <div
            class="img__part"
          />
        </div>
      </div>
      <div
        class="container__sell__item"
      >
        <div
          class="container__img__button"
        >
          <div
            class="img__const"
          />
          <div
            class="button__sell"
          />
        </div>
        <div
          class="text__sell"
        >
          <h3>
            Создайте велосипед своей мечты с нашим конструктором! Хотите уникальный велосипед, который отражает ваш стиль и нужды? Наш конструктор позволяет выбрать все — от рамы до последних штрихов. Погрузитесь в процесс создания и получите велосипед, который будет идеально соответствовать вашим ожиданиям. Перейдите на страницу конструктора и начните своё уникальное путешествие!
          </h3>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
